pkgs/applications/audio/mpg321/default.nix

```diff
@@ -12,6 +16,8 @@ stdenv.mkDerivation rec {
   hardeningDisable = [ "format" ];
 
   configureFlags = [
+    (stdenv.lib.optional (defaultAudio != null)
+      "--with-default-audio=${defaultAudio}")
```

Comment:
The `optional` function returns a list so I think this actually should be something like
```
  configureFlags = [
    ("--enable-alsa=" + (if stdenv.isLinux then "yes" else "no"))
  ] ++ stdenv.lib.optional (defaultAudio != null)
    "--with-default-audio=${defaultAudio}";
```

Reply:
Okay, huh, I have no idea how it worked then. It clearly did. Probably something bash interpreted correctly by chance.

Reply:
Yes, it correctly sets the configure flag. I changed it to a correct list concatenation anyway.